services/121-service/src/transaction-jobs/services/transaction-jobs-helper.service.ts

```diff
@@ -128,4 +140,110 @@ export class TransactionJobsHelperService {
       userId,
     });
   }
+
+  public async saveTransactionProgressAndUpdateRelatedData({
+    newTransactionStatus,
+    context,
+    description,
+    errorMessage,
+  }: {
+    newTransactionStatus: TransactionStatusEnum;
+    context: SaveTransactionProgressAndRelatedDataContext;
+    description: TransactionEventDescription;
+    errorMessage?: string;
+  }): Promise<void> {
+    if (!context.isRetry) {
+      await this.updatePaymentCountAndSetToCompleted({
+        referenceId: context.referenceId,
+        programId: context.programId,
+        userId: context.userId,
+      });
+    }
+
+    await this.transactionsService.saveTransactionProgress({
+      context,
+      description,
+      errorMessage,
+      newTransactionStatus,
+    });
+  }
```

Comment:
This function and the context parameter are really hard to name.

I was thinking about this:
```
transanctionJobHelper.saveTransactionProgress // Updating payment count and completion status can be seen as transactionProgress I think


transactionsService.saveProgress // It's nice if this function has a different name than the one on the jobhelper and I think transaction does not have to be in there because of class name

```

Reply:
saveTransactionProgress > saveTransactionProgressAndUpdateRegistration?